File: src/leetcode/2026/03/Solution717.java
public class Solution717 {
    public boolean isOneBitCharacter(int[] bits) {
        int n = bits.length;
        int i = 0;

        while (i < n - 1) {
            if (bits[i] == 1) {
                i += 2; // Skip the next bit as well
            } else {
                i += 1; // Move to the next bit
            }
        }

        return i == n - 1; // Check if we ended on the last bit
    }

    public static void main(String[] args) {
        Solution717 solution = new Solution717();

        int[] input = { 1, 0, 0 };
        boolean output = solution.isOneBitCharacter(input);
        System.out.println("Output: " + output);

        input = new int[] { 1, 1, 1, 0 };
        output = solution.isOneBitCharacter(input);
        System.out.println("Output: " + output);

        input = new int[] { 0 };
        output = solution.isOneBitCharacter(input);
        System.out.println("Output: " + output);
    }
}

File: src/leetcode/2026/03/Solution717.cpp
#include <vector>
#include <iostream>
using namespace std;

class Solution717 {
public:
    bool isOneBitCharacter(vector<int>& bits) {
        int n = bits.size();
        int i = 0;

        while (i < n - 1) {
            if (bits[i] == 1) {
                i += 2; // Skip the next bit as well
            } else {
                i += 1; // Move to the next bit
            }
        }
        return i == n - 1;
    }
};

int main() {
    Solution717 solution;
    
    vector<int> bits1 = {1, 0, 0};
    vector<int> bits2 = {1, 1, 1, 0};
    
    bool result1 = solution.isOneBitCharacter(bits1);
    bool result2 = solution.isOneBitCharacter(bits2);
    
    cout << "Input: [1, 0, 0] -> Output: " << (result1 ? "true" : "false") << endl; // Should print true
    cout << "Input: [1, 1, 1, 0] -> Output: " << (result2 ? "true" : "false") << endl; // Should print false
}